tailcfg/tailcfg.go

```diff
@@ -352,10 +353,16 @@ type Node struct {
 	// a QWERTY keyboard (3.3.40)). The "port number" is the home DERP region ID
 	// integer.
 	//
-	// TODO(bradfitz): simplify this legacy mess; add a new HomeDERPRegionID int
-	// field behind a new capver bump.
+	// Deprecated: HomeDERP has replaced this, but old servers might still send
+	// this field. See tailscale/tailscale#14636. Do not use this field in code
+	// other than in the upgradeNode func, which canonicalizes it to HomeDERP
+	// if it arrives as a DERP string on the wire.
 	DERP string `json:",omitempty"` // DERP-in-IP:port ("127.3.3.40:N") endpoint
 
+	// HomeDERP is the modern version of the DERP string field, with just
+	// an integer. The client advertises support for this as of capver 111.
+	HomeDERP int `json:",omitempty"` // DERP region ID of the node's home DERP
```

Comment:
Might be worth explicitly stating what `0` means?

Reply:
done